### PR TITLE
Project files improvement and more target-frameworks

### PR DESCRIPTION
**Pull request by moien007**
Hi,

This PR contains the following changes:
- Improvement and simplification of projects files
- Updating NuGet package references to the latest available
- Support for .NET Framework 4.8
- Support for .NET Standard 2.1
- Using [`BitOperations`](https://docs.microsoft.com/en-us/dotnet/api/system.numerics.bitoperations?view=netcore-3.1) class for .NET Core 3.1 ([The explanation](https://github.com/uranium62/xxHash/commit/7b907932cd9921e150dad210e80a199d1cfcf384))

These changes are not what I supposed to make but I have to commit that I may not be able to solve the two issues #40 and #41 at all due to my lack of knowledge of the NaCl itself, sorry for that.

Regards.

### CODE REVIEW

File: src/NaCl.Core/NaCl.Core.csproj [hunk 1]
```diff
@@ -1,8 +1,9 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard1.3;netstandard2.0</TargetFrameworks>
-    <TargetFrameworks Condition="'$(OS)' != 'Unix'">net45;net47;netstandard1.3;netstandard2.0</TargetFrameworks>
```
Comment (idaviddesmet) [inline src/NaCl.Core/NaCl.Core.csproj]: `<TargetFrameworks Condition="'$(OS)' != 'Unix'">net45;net47;netstandard1.3;netstandard2.0</TargetFrameworks>`
The above conditional is for supporting other platforms other than Windows, if we leave it just as `<TargetFrameworks>netstandard1.3;netstandard2.0;netstandard2.1;netcoreapp3.1;net45;net48</TargetFrameworks>` it won't compile on Linux and macOS.

File: src/NaCl.Core/NaCl.Core.csproj [hunk 2]
```diff
@@ -16,21 +17,12 @@
...
-    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <LangVersion>latest</LangVersion>
-  </PropertyGroup>
-
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
-    <LangVersion>latest</LangVersion>
-  </PropertyGroup>
-
-  <Target Name="LogDebugInfo">
-    <Message Text="Building for $(TargetFramework) on $(OS)" Importance="High" />
```
Comment (idaviddesmet) [inline src/NaCl.Core/NaCl.Core.csproj]: This message is useful for debugging purposes on CI Pipeline.

File: src/NaCl.Core.Tests/NaCl.Core.Tests.csproj [hunk 3]
```diff
@@ -1,45 +1,28 @@
...
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
   </ItemGroup>
 
   <ItemGroup>
     <ProjectReference Include="..\NaCl.Core\NaCl.Core.csproj" />
   </ItemGroup>
-
-  <Target Name="LogDebugInfo">
-    <Message Text="Building for $(TargetFramework) on $(OS)" Importance="High" />
```
Comment (idaviddesmet) [inline src/NaCl.Core.Tests/NaCl.Core.Tests.csproj]: This message is useful for debugging purposes on CI Pipeline.

File: src/NaCl.Core/NaCl.Core.csproj [hunk 4]
```diff
@@ -1,8 +1,9 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard1.3;netstandard2.0</TargetFrameworks>
-    <TargetFrameworks Condition="'$(OS)' != 'Unix'">net45;net47;netstandard1.3;netstandard2.0</TargetFrameworks>
+    <TargetFrameworks>netstandard1.3;netstandard2.0;netstandard2.1;netcoreapp3.1;net45;net48</TargetFrameworks>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+
```
Comment (idaviddesmet) [inline src/NaCl.Core/NaCl.Core.csproj]: Should add back `<LangVersion>latest</LangVersion>` below `<AllowUnsafeBlocks>true</AllowUnsafeBlocks>` 😉

File: src/NaCl.Core/NaCl.Core.csproj (hunk 1 above)
Comment (idaviddesmet) [reply src/NaCl.Core/NaCl.Core.csproj]: Suggested change:
```
<TargetFrameworks>netstandard1.3;netstandard2.0;netstandard2.1;netcoreapp3.1</TargetFrameworks>
<TargetFrameworks Condition="'$(OS)' != 'Unix'">netstandard1.3;netstandard2.0;netstandard2.1;netcoreapp3.1;net45;net48</TargetFrameworks>
```

Comment (moien007) [reply src/NaCl.Core/NaCl.Core.csproj]: Sorry, I didn't know that.

**Comment by moien007** [comment]
I'm a noob in git. I addressed your requested changes by pushing [this commit](https://github.com/idaviddesmet/NaCl.Core/pull/42/commits/0dadabcf7e54f507554a2827a581ecdf462158de), is that how I should have done that?

**Comment by idaviddesmet** [comment]
> I'm a noob in git. I addressed your requested changes by pushing [this commit](https://github.com/idaviddesmet/NaCl.Core/pull/42/commits/0dadabcf7e54f507554a2827a581ecdf462158de), is that how I should have done that?

No worries, you did it right! We all started as noob in git. 😉
Will merge your PR and address the CI Pipeline later since it's giving some errors.

File: src/NaCl.Core/NaCl.Core.csproj (hunk 1 above)
Comment (idaviddesmet) [reply src/NaCl.Core/NaCl.Core.csproj]: No biggie, that's why there are reviews ;)

**Comment by idaviddesmet** [comment]
@moien007 The changes will be pushed later to nuget.org but in case you feel eager to try it now, you can grab them from [here](https://dev.azure.com/idaviddesmet/NaCl.Core/_packaging?_a=feed&feed=NaCl.Core-CI).

**Pull request merged**